### PR TITLE
[Feature/APP-103] new setting: enforce native installer

### PR DESCRIPTION
**Pull request by jdandrade**
**What does this PR do?**

   new setting - enforce native installer. This will allow Xiaomi users to recover from the state of not being able to install any application in the app because of the PackageInstaller.

EDIT: Strings on this branch are on temporary. The ones from dev are the correct ones.

**Database changed?**

  No

**Where should the reviewer start?**

- [ ] settings.xml
- [ ] AppBundlesVisibilityManager.java
- [ ] InstallManager.java

**How should this be manually tested?**

  - Test installations with both setting on/off on Xiaomi and non-xiaomi devices.

**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-103](https://aptoide.atlassian.net/browse/APP-103)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/install/InstallManager.java [hunk 1]
```diff
@@ -240,19 +241,25 @@ private boolean isAppInstalling(List<RoomInstalled> installingAppList, String pa
   }
 
   public Completable install(RoomDownload download) {
-    return install(download, false, false, true);
+    boolean shouldForceDefaultInstall =
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/install/InstallManager.java:244]: Why not use the settingsProvider you created here? You have a method for isEnforceNativeInstaller() and you wouldn't need to extract the variable either

File: app/src/main/java/cm/aptoide/pt/install/InstallManager.java [hunk 2]
```diff
@@ -240,19 +241,25 @@ private boolean isAppInstalling(List<RoomInstalled> installingAppList, String pa
...
+        sharedPreferences.getBoolean(ManagedKeys.ENFORCE_NATIVE_INSTALLER_KEY, false);
+    return install(download, shouldForceDefaultInstall, false, true);
   }
 
   public Completable install(RoomDownload download, boolean shouldInstall) {
-    return install(download, false, false, shouldInstall);
+    boolean shouldForceDefaultInstall =
+        sharedPreferences.getBoolean(ManagedKeys.ENFORCE_NATIVE_INSTALLER_KEY, false);
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/install/InstallManager.java:251]: same as above

File: app/src/main/java/cm/aptoide/pt/install/InstallManager.java [hunk 3]
```diff
@@ -240,19 +241,25 @@ private boolean isAppInstalling(List<RoomInstalled> installingAppList, String pa
...
   }
 
   private Completable defaultInstall(RoomDownload download) {
     return install(download, true, false, true);
   }
 
   public Completable splitInstall(RoomDownload download) {
-    return install(download, false, true, true);
+    boolean shouldForceDefaultInstall =
+        sharedPreferences.getBoolean(ManagedKeys.ENFORCE_NATIVE_INSTALLER_KEY, false);
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/install/InstallManager.java:261]: same as above

Comment (phillipsgoncalves) [reply app/src/main/java/cm/aptoide/pt/install/InstallManager.java:261]: Also here, note for the future, but forceSplitInstall and forceDefaultInstall are mutually exclusive as we can only use one type of install for each install.

**Review by phillipsgoncalves — CHANGES_REQUESTED** [review]

**Comment by phillipsgoncalves** [comment]
After some tests, I noticed the following:
- after changing the installer setting should we refresh the updates automatically? This is important because after changing the setting, if we navigate to the apps section, we will still have old updates/updates with app bundles which will not be installed.
- some updates are stuck on installing even after the installation finished (wether with app not installed or success) (this happened on my pixel 5, could not reproduce on my xiami as on my xiaomi device i could not see those apps)

**Comment by jdandrade** [comment]
> * after changing the installer setting should we refresh the updates automatically? This is important because after changing the setting, if we navigate to the apps section, we will still have old updates/updates with app bundles which will not be installed.

Good question. This feature is to be a fallback to a really bad experience, a lifeboat for users stuck on this situation. Even if the UX is not optimal this users will most likely be in contact with us about the issue so we can just tell them to refresh/clear data on the device.

**Pull request merged**